cloudbridge/cloud/providers/azure/azure_client.py

```diff
@@ -278,6 +290,39 @@ def create_storage_account(self, name, params):
         return self.storage_client.storage_accounts. \
             create(self.resource_group, name.lower(), params).result()
 
+    # Create a storage account. To prevent a race condition, try
+    # to get or create at least twice
+    @tenacity.retry(stop=tenacity.stop_after_attempt(2),
+                    retry=tenacity.retry_if_exception_type(CloudError),
+                    reraise=True)
+    def _get_or_create_storage_account(self):
+        if self._storage_account:
+            return self._storage_account
+        else:
+            try:
+                self._storage_account = \
+                    self.get_storage_account(self.storage_account)
+            except CloudError as cloud_error:
+                if "not found" in cloud_error.message:
+                    storage_account_params = {
+                        'sku': {
+                            'name': 'Standard_LRS'
+                        },
+                        'kind': 'storage',
+                        'location': self.region_name,
+                    }
+                    self._storage_account = \
+                        self.create_storage_account(self.storage_account,
+                                                    storage_account_params)
+                elif "is already taken" in cloud_error.message:
```

Comment:
Won't this exception be raised only when creating a storage account, not when retrieving one?

Reply:
Yes, this part wasn't finished, sorry about that. I was just pushing to the new branch before bed to launch a travis test so I don't have to leave my laptop open or wait for the tests to run before going to sleep.